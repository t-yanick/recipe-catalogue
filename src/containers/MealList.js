import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Grid, Container } from 'semantic-ui-react';
import Button from 'react-bootstrap/Button';
import { CardColumns } from 'react-bootstrap';
import Card from 'react-bootstrap/Card';
import { fetchMealsStartAsync } from '../actions/getMeals';

const MealList = (props) => {
  const { cat } = props;
  const meals = useSelector((state) => state.meals.meals);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMealsStartAsync(cat));
  }, [cat]);

  return (
    <Container textalign="center" style={{ marginTop: 40 }}>
      <Grid>
        <Grid.Column className="d-flex flex-wrap mx-5 col-md-12">
          {
            meals && meals.map((meal) => (
              <CardColumns
                key={meal.strMeal}
                style={{ marginTop: 20 }}
                as={Link}
                to={{
                  pathname: `/ingredients/${meal.idMeal}`,
                  id: meal.idMeal,
                }}
              >
                <Card className="m-1 mx-4 pb-1" style={{ width: '18rem' }}>
                  <Card.Img variant="top" src={meal.strMealThumb} />
                  <Card.Body>
                    <Card.Title>
                      {meal.strMeal}
                    </Card.Title>
                    <Button variant="primary">
                      Get Recipe
                    </Button>
                  </Card.Body>
                </Card>
              </CardColumns>
            ))
          }
        </Grid.Column>
      </Grid>
    </Container>
  );
};

MealList.propTypes = {
  cat: PropTypes.string,
};

MealList.defaultProps = {
  cat: '',
};

export default MealList;
