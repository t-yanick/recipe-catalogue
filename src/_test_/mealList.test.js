import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import MealList from '../containers/MealList';

describe('Creates a component', () => {
  const middlewares = [thunk];
  const initialState = { meals: ['chicken'] };
  const mockStore = configureStore(middlewares);
  let store;
  it('renders correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <Provider store={store}>
        <MealList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});