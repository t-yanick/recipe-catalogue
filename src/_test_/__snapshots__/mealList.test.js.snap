// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": 40,
    }
  }
  textalign="center"
>
  <div
    className="ui grid"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
